File: app/src/main/java/com/xuie/android/ui/recycler/discrete/DiscreteFragment.java
package com.xuie.android.ui.recycler.discrete;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.RecyclerView;

import com.xuie.android.R;
import com.yarolegovich.discretescrollview.DiscreteScrollView;
import com.yarolegovich.discretescrollview.InfiniteScrollAdapter;
import com.yarolegovich.discretescrollview.Orientation;
import com.yarolegovich.discretescrollview.transform.ScaleTransformer;

import java.util.List;

/**
 * A simple {@link Fragment} subclass.
 *
 * @author xuie
 */
public class DiscreteFragment extends Fragment implements DiscreteScrollView.OnItemChangedListener<RecyclerView.ViewHolder> {
    private static final String TAG = "DiscreteFragment";

    private InfiniteScrollAdapter infiniteAdapter;

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_discrete, container, false);

        DiscreteScrollView itemPicker = view.findViewById(R.id.item_picker);

        Shop shop = Shop.get();
        List<Item> data = shop.getData();
        itemPicker.setOrientation(Orientation.HORIZONTAL);
        itemPicker.addOnItemChangedListener(this);
        infiniteAdapter = InfiniteScrollAdapter.wrap(new ShopAdapter(data));
        itemPicker.setAdapter(infiniteAdapter);
        itemPicker.setSlideOnFling(true);
        itemPicker.setItemTransitionTimeMillis(DiscreteScrollViewOptions.getTransitionTime());
        itemPicker.setItemTransformer(new ScaleTransformer.Builder()
                .setMinScale(0.8f)
                .build());

        return view;
    }

    @Override
    public void onCurrentItemChanged(@Nullable RecyclerView.ViewHolder viewHolder, int adapterPosition) {
        int positionInDataSet = infiniteAdapter.getRealPosition(adapterPosition);
    }
}

File: app/src/main/java/xuk/android/ui/recycler/discrete/DiscreteFragment.kt
package xuk.android.ui.recycler.discrete

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.RecyclerView
import com.yarolegovich.discretescrollview.DiscreteScrollView
import com.yarolegovich.discretescrollview.InfiniteScrollAdapter
import com.yarolegovich.discretescrollview.Orientation
import com.yarolegovich.discretescrollview.transform.ScaleTransformer
import kotlinx.android.synthetic.main.fragment_discrete.*
import xuk.android.R

class DiscreteFragment : Fragment(R.layout.fragment_discrete),
    DiscreteScrollView.OnItemChangedListener<RecyclerView.ViewHolder> {

  override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
    super.onViewCreated(view, savedInstanceState)

    val shop = Shop.get()
    val data = shop.data
    item_picker.apply {
      setOrientation(Orientation.HORIZONTAL)
      addOnItemChangedListener(this@DiscreteFragment)
      adapter = InfiniteScrollAdapter.wrap(ShopAdapter(data))

      setSlideOnFling(true)
      setItemTransitionTimeMillis(DiscreteScrollViewOptions.transitionTime)
      setItemTransformer(ScaleTransformer.Builder()
          .setMinScale(0.8f)
          .build())
    }
  }

  override fun onCurrentItemChanged(viewHolder: RecyclerView.ViewHolder?, adapterPosition: Int) {

  }
}
